protoarray/src/test/java/tech/pegasys/artemis/protoarray/ComputeDeltasTest.java

```diff
@@ -0,0 +1,277 @@
+
+    for (int i = 0; i < deltas.size(); i++) {
+      long delta = deltas.get(i);
+      if (i == 0) {
+        // Zero'th root should have a negative delta
+        assertThat(delta).isEqualTo(-totalDelta);
+      } else if (i == 1) {
+        // First root should have positive delta
+        assertThat(delta).isEqualTo(totalDelta);
+      } else {
+        // All other deltas should be zero
+        assertThat(delta).isEqualTo(0L);
+      }
+    }
+
+    votesShouldBeUpdated(votes);
+  }
+
+  @Test
+  void moveOutOfTree() {
+    final UnsignedLong BALANCE = valueOf(42);
+
+    // There is only one block.
+    indices.put(getHash(1), 0);
+
+    // There are two validators.
+    oldBalances = Collections.nCopies(2, BALANCE);
+    newBalances = Collections.nCopies(2, BALANCE);
+
+    // One validator moves their vote from the block to the zero hash.
+    votes.add(new VoteTracker(getHash(1), Bytes32.ZERO, ZERO));
+
+    // One validator moves their vote from the block to something outside the tree.
+    votes.add(new VoteTracker(getHash(1), getHash(1337), ZERO));
+
+    List<Long> deltas = computeDeltas(indices, votes, oldBalances, newBalances);
+    assertThat(deltas).hasSize(1);
+
+    // The block should have lost both balances
+    assertThat(deltas.get(0)).isEqualTo(-BALANCE.longValue() * 2);
+
+    votesShouldBeUpdated(votes);
+  }
+
+  @Test
+  void changingBalances() {
+
+    final UnsignedLong OLD_BALANCE = valueOf(42);
+    final UnsignedLong NEW_BALANCE = OLD_BALANCE.times(valueOf(2));
+
+    int validatorCount = 16;
+
+    for (int i = 0; i < validatorCount; i++) {
+      indices.put(getHash(i), i);
+      votes.add(new VoteTracker(getHash(0), getHash(1), ZERO));
+      oldBalances.add(OLD_BALANCE);
+      newBalances.add(NEW_BALANCE);
+    }
+
+    List<Long> deltas = computeDeltas(indices, votes, oldBalances, newBalances);
+    assertThat(deltas).hasSize(validatorCount);
+
+    for (int i = 0; i < deltas.size(); i++) {
+      long delta = deltas.get(i);
+      if (i == 0) {
+        // Zero'th root should have a negative delta
+        assertThat(delta).isEqualTo(-OLD_BALANCE.longValue() * validatorCount);
+      } else if (i == 1) {
+        // First root should have positive delta
+        assertThat(delta).isEqualTo(NEW_BALANCE.longValue() * validatorCount);
+      } else {
+        // All other deltas should be zero
+        assertThat(delta).isEqualTo(0L);
+      }
+    }
+
+    votesShouldBeUpdated(votes);
+  }
+
+  @Test
+  void validatorAppears() {
+    final UnsignedLong BALANCE = valueOf(42);
+
+    // There are two blocks.
+    indices.put(getHash(1), 0);
+    indices.put(getHash(2), 1);
+
+    // There is only one validator in the old balances.
+    oldBalances.add(BALANCE);
+
+    // There are two validators in the new balances.
+    newBalances.addAll(List.of(BALANCE, BALANCE));
+
+    // Both validators move votes from block 1 to block 2.
+    for (int __ = 0; __ < 2; __++) {
```

Comment:
We should use a real variable name here not just `__` - we really are using the value if only to check when to exit the loop.
```suggestion
    for (int i = 0; i < 2; i++) {
```

Reply:
Done. Honestly, I don't know what had gotten into me to write it that way.